### PR TITLE
Add some API tests for TLSv1.3 record padding

### PR DESCRIPTION
**Pull request by mattcaswell**
We have some ssl_test_new tests for record padding. But these only use
the block padding approach set via a config file on the SSL_CTX. We add
tests for all the various API calls.

### CODE REVIEW

**Comment by t8m** [comment]
CI is relevant

**Comment by mattcaswell** [comment]
> CI is relevant

Fixup pushed.

File: test/sslapitest.c [hunk 1]
```diff
@@ -9959,6 +9959,127 @@ static int test_read_ahead_key_change(void)
...
+
+    if (!TEST_true(create_ssl_connection(serverssl, clientssl, SSL_ERROR_NONE)))
+        goto end;
+
+    called = 0;
+    /*
+     * Write some data, then check we can read it. Do this four times to check
+     * we can continue to write and read padded data after the initial record
+     * padding has been added. We don't actually check that the padding has
+     * been applied to the record - just that we can continue to communicte
```
Comment (t8m) [inline test/sslapitest.c:10054]: Typo: communicate

**Review by t8m — APPROVED** [review]
Just one nit to fix

File: test/sslapitest.c (hunk 1 above)
Comment (mattcaswell) [reply test/sslapitest.c]: Fixed

**Review by t8m — APPROVED** [review]

**Comment by FdaSilvaYY** [comment]
Typo in commit message, and so in PR description : the the

**Comment by mattcaswell** [comment]
> Typo in commit message, and so in PR description : the the

Fixed

**Comment by t8m** [comment]
Still OK

**Review by paulidale — APPROVED** [review]

**Comment by hlandau** [comment]
Merged to master. Thank you.

**Pull request closed without merging**